File: online-judge-solutions/Codeforces/1670A.cpp
//https://codeforces.com/problemset/problem/1670/A
//Lang		:	GNU C++14
//Time		:	31 ms
//Memory	:	800 KB
#include<iostream>
#include<vector>
#include<algorithm>
using namespace std;
#define fastInp cin.tie(0); cout.tie(0); ios_base::sync_with_stdio(0);
int main(){
	fastInp;
	int t,n;
	for(cin>>t;t--;){
		cin>>n;
		vector<int>v(n),w;
		int z=0;
		for(int i=0;i<n;i++){
			cin>>v[i];
			if(v[i]<0) z++;
		}
		for(int i=0;i<n;i++){
			v[i]=(i<z?-abs(v[i]):abs(v[i]));
		}
		w=v;
		sort(begin(w),end(w));
		cout<<(v==w?"YES":"NO")<<"\n";
	}
	return 0;
}

File: online-judge-solutions/Codeforces/1670A.py
#https://codeforces.com/problemset/problem/1670/A
#Lang		:	Python 3.8
#Time		:	186 ms
#Memory         :	7500 KB
for _ in range(int(input())):
    n,z,w=int(input()),0,[]
    v=list(map(int,input().split()))[:n]
    z=[z+1 for i in v if(i<0)]
    for i in range(n):
        v[i]=(-abs(v[i]) if i<len(z) else abs(v[i]))
    w=v[:]
    v.sort()
    print("YES" if v==w else "NO")
